setup/setup.js

```diff
@@ -1,93 +1,182 @@
                 }
 
-                status('Enabling PubNub features...');
+                let app = response.result;
 
-                let result = false;
+                status('Getting PubNub keys...');
 
-                data.result.forEach((res) => {
-                    if(res.id == created.result.id) {
-                        result = res;
+                api.request('get', ['api', 'apps'], {
+                    data: {
+                        owner_id: account.id
                     }
-                });
+                }, (err, response) => {
 
-                let key = result.keys[0];
-
-                key.properties.name = 'ChatEngine Keyset';
-                key.properties.presence = 1;
-                key.properties.history = 1;
-                key.properties.message_storage_ttl = 7;
-                key.properties.multiplexing = 1;
-                key.properties.presence_announce_max = 20;
-                key.properties.presence_debounce = 2;
-                key.properties.presence_global_here_now = 1;
-                key.properties.presence_interval = 10;
-                key.properties.presence_leave_on_disconnect = 0;
-                key.properties.blocks = 1;
-                key.properties.uls = 1;
-                key.properties.wildcardsubscribe = 1;
-
-                api.request('put', ['api', 'keys', key.id], {
-                    data: key
-                }, function (err, data) {
-
-                    if(err) {
-                        return callback('Could not create a new key. Please contact support@pubnub.com.');
+                    if (err) {
+                        return callback('Could not get PubNub keys. Please contact support@pubnub.com.');
                     }
 
-                    status('Done!');
+                    let apps = response.result;
+                    let key;
 
-                    callback(null, data);
+                    for (let item of apps) {
+                        if (item.id == app.id) {
+                            key = item.keys[0];
+                        }
+                    }
 
+                    status('Enabling PubNub features...');
+
+                    key.properties.name = 'ChatEngine Keyset';
+                    key.properties.presence = 1;
+                    key.properties.history = 1;
+                    key.properties.message_storage_ttl = 7;
+                    key.properties.multiplexing = 1;
+                    key.properties.presence_announce_max = 20;
+                    key.properties.presence_debounce = 2;
+                    key.properties.presence_global_here_now = 1;
+                    key.properties.presence_interval = 10;
+                    key.properties.presence_leave_on_disconnect = 0;
+                    key.properties.blocks = 1;
+                    key.properties.uls = 1;
+                    key.properties.wildcardsubscribe = 1;
```

Comment:
Oh yeah, these settings need to be reviewed by product. No required changes on your end though.